packages/fxa-auth-server/lib/routes/account.ts

```diff
@@ -1512,27 +1512,31 @@ export class AccountHandler {
     }
 
     // In other scenarios, fall back to the default behavior and let the user
-    // delete the account
-    const password = new this.Password(
-      authPW,
-      emailRecord.authSalt,
-      emailRecord.verifierVersion
-    );
+    // delete the account. If they have a password set, we verify it here. Users
+    // that don't have a password set will be able to delete their account without
+    // this step.
+    if (accountRecord.veriferSetAt > 0) {
+      const password = new this.Password(
```

Comment:
nit: Not technically wrong, but there's some about `new this.` that seems strange to me. It's just uncommon to see an instance being created from a function on an instance.